File: moviejava/app/src/main/java/com/example/moviejava/data/remote/film/FilmRepository.java
package com.example.moviejava.data.remote.film;


import com.example.moviejava.model.casts.CastsFilm;
import com.example.moviejava.model.film.Film;
import com.example.moviejava.model.film.video.Video;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.Retrofit;

@Singleton
public class FilmRepository {
    @Inject
    public FilmRepository(Retrofit retrofit) {
        this.service = retrofit.create(FilmService.class);
    }

    private FilmService service;

    public Observable<Film> getFilm(int id) {
        return service.getFilm(id);
    }

    public Observable<Video> getVideo(int id) {
        return service.getVideo(id);
    }

    public Observable<CastsFilm>getCast(int id) {
        return service.getCastFilm(id);
    }
}

File: movie/app/src/main/java/com/example/movie/data/remote/film/FilmRepository.kt
package com.example.movie.data.remote.film


import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.flow
import kotlinx.coroutines.flow.flowOn
import retrofit2.Retrofit
import javax.inject.Inject

class FilmRepository @Inject constructor(val retrofit: Retrofit) {
    private val service = retrofit.create(FilmService::class.java)
    fun getFilm(id: Int) = flow {
        emit(
            service.getFilm(id = id)
        )
    }.flowOn(Dispatchers.IO)

    fun getVideo(id: Int) = flow {
        emit(
            service.getVideo(id = id)
        )
    }.flowOn(Dispatchers.IO)

    fun getCastFilm(id: Int) = flow {
        emit(service.getCastFilm(id = id))
    }.flowOn(Dispatchers.IO)
}